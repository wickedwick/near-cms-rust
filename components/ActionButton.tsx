import React from 'react'

const ActionButton = ({ children, onClick }: { children: React.ReactNode; onClick: () => void }) => {
  return (
    <button
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
      onClick={onClick}
    >
      {children}
    </button>
  )
}

export default ActionButton
