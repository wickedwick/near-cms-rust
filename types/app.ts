export type NearUser = {
  accountId: string
  balance: string
}

export type ContentType = {
  name: string
  description: string
  fields: string[]
}

export type Content = {
  name: string
  slug: string
  content_type: ContentType
  is_public: boolean
  is_encrypted: boolean
  owner: string
}

export type UserRole = {
  account_id: string
  role: number
}
