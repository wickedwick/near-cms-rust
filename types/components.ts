import { Dispatch, SetStateAction } from "react"

export type InputGroupProps = {
  for: string
  groupClassName: string
  inputClassName: string
  label: string
  labelClassName: string
  placeholder: string
  type: "text" | "number" | "password" | "email" | "tel" | "date" | "time" | "datetime-local" | "month" | "week" | "url" | "search" | "color" | "file"
  value: string
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
}

export type CheckboxProps = {
  checked: boolean,
  groupClassName: string
  inputClassName: string
  label: string
  labelClassName: string
  name: string,
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
}

export type Field = {
  fieldType: string
  name: string
  value: string
  required: boolean
  maxLength: number
}

export type FieldTypesEditorProps = {
  fields: Field[],
  setFields: Dispatch<SetStateAction<Field[]>>
}

export type Option = {
  label: string
  value: string
}

export const fieldTypeOptions: Option[] = [
  { label: 'String', value: 'String' },
  { label: 'RichText', value: 'RichText' },
  { label: 'Int', value: 'Int' },
  { label: 'Float', value: 'Float' },
  { label: 'Boolean', value: 'Boolean' },
  { label: 'Date', value: 'Date' },
  { label: 'DateTime', value: 'DateTime' },
  { label: 'Time', value: 'Time' },
  { label: 'Password', value: 'Password' },
  { label: 'Email', value: 'Email' },
  { label: 'URL', value: 'URL' },
  { label: 'Phone', value: 'Phone' },
  { label: 'Postal Code', value: 'PostalCode' },
  { label: 'IP', value: 'IP' },
  { label: 'Image', value: 'Image' },
  { label: 'Video', value: 'Video' },
  { label: 'File', value: 'File' },
  { label: 'Custom', value: 'Custom' }
]

export type LayoutProps = {
  children: React.ReactNode
  home: boolean
}
