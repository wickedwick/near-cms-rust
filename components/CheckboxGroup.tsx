import React from 'react'
import { CheckboxProps } from '../types/components'

const CheckboxGroup = (props: CheckboxProps) => {
  return (
    <div className={`md:flex md:items-center mb-6 ${props.groupClassName}`}>
      <label className={`md:w-2/3 block text-gray-500 font-bold ${props.labelClassName}`}>
        <input className={`mr-2 leading-tight ${props.inputClassName}`}
          checked={props.checked}
          name={props.name}
          onChange={props.onChange}
          type="checkbox"
        />
        <span className="text-sm">
          {props.label}
        </span>
      </label>
    </div>
  )
}

export default CheckboxGroup
