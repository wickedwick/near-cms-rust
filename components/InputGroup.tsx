import React from 'react'
import { InputGroupProps } from '../types/components'

const InputGroup = (props: InputGroupProps) => {
  return (
    <div className="mb-4">
      <label
        className={`block text-gray-700 text-sm font-bold mb-2 ${props.labelClassName}`}
        htmlFor={props.for}
      >
        {props.label}
      </label>
      <input
        className={`shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline ${props.inputClassName}`}
        id={props.for}
        onChange={props.onChange}
        placeholder={props.placeholder}
        type={props.type}
        value={props.value}
      />
    </div>
  )
}

export default InputGroup
