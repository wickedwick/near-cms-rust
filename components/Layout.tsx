import Head from 'next/head';
import Link from 'next/link';
import Nav from './Nav';
import { LayoutProps } from '../types/components';

const Layout = ({ children, home }: LayoutProps): JSX.Element => {
  return (
    <>
      <Head>
        <title>d CMS</title>
        <meta name="description" content="A decentralized content management system." />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Nav />
      <div className="">
        <main>{children}</main>
        {!home && (
          <div className="">
            <Link href="/">
              <a>‚Üê Back to home</a>
            </Link>
          </div>
        )}
      </div>
    </>
  )
}

export default Layout
