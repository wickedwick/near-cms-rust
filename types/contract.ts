import * as nearAPI from 'near-api-js'
import { Content, ContentType, NearUser, UserRole } from './app'

export type NearContract = {
  contract: nearAPI.Contract
  currentUser: NearUser | undefined
  walletConnection: nearAPI.WalletConnection
}

export interface CmsContract extends nearAPI.Contract {
  // getUsers: () => Promise<UserRole[]>
  get_user_role: ({ account_id }: { account_id: string }) => Promise<UserRole>
  set_user_role: ({ account_id, user_to_add, role }: { account_id: string, user_to_add: string, role: number }) => Promise<void>
  get_content: ({ slug }: { slug: string }) => Promise<Content>
  set_content: ({ name, content_type_name, is_public, is_encrypted}: { name: string, content_type_name: string, is_public: boolean, is_encrypted: boolean }) => Promise<void>
  //deleteContent: ({ content }: { content: Content }) => Promise<void>
  // getMediaBySlug: ({ slug }: { slug: string }) => Promise<Media>
  // getMedia: () => Promise<Media[]>
  // setMedia: ({ args, callbackUrl }: { args: { media: Media }, callbackUrl: string }) => Promise<void>
  //deleteMedia: ({ args, callbackUrl }: { args: { slug: string }, callbackUrl: string }) => Promise<void>
  get_contents: () => Promise<Content[]>
  // getPublicContent: ({ slug }: { slug: string }) => Promise<Content>
  // getPublicContents: () => Promise<Content[]>
  // getClient: ({ slug }: { slug: string }) => Promise<Client>
  // setClient: ({ args, callbackUrl }: { args: { clientUpdate: Client }, callbackUrl: string }) => Promise<void>
  // getClients: () => Promise<Client[]>
  set_content_type: ({ name, description, fields}: {name: string, description: string, fields: string[]}) => Promise<void>
  get_content_type: ({ name }: { name: string }) => Promise<ContentType>
  get_content_types: () => Promise<ContentType[]>
  // deleteContentType: ({ name }: { name: string }) => Promise<void>
  // getUserRole: ({ name }: { name: string }) => Promise<UserRole>
}