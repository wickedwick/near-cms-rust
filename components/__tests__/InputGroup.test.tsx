import React from 'react'
import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import InputGroup from '../InputGroup'
import { InputGroupProps } from '../../types/components';

configure({ adapter: new Adapter() });

const onChange = jest.fn()
const props: InputGroupProps = {
  groupClassName: '',
  labelClassName: '',
  inputClassName: '',
  for: '',
  label: '',
  value: '',
  placeholder: '',
  type: 'text',
  onChange
}

describe('InputGroup', () => {
  it('should render', () => {
    const wrapper = shallow(<InputGroup {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call onChange', () => {
    const wrapper = shallow(<InputGroup {...props} />)
    wrapper.find('input').simulate('change')
    expect(onChange).toHaveBeenCalled()
  })
})
