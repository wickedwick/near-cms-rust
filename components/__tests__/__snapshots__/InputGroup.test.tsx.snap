// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render 1`] = `ShallowWrapper {}`;
